draft-ietf-jsonpath-base.md

```diff
@@ -1551,23 +1634,23 @@ string matches a given regular expression, which is in {{-iregexp}} form.
 $[?match(@.date, "1974-05-..")]
 ~~~
 
-Its first argument is an optional string that is matched against the iregexp
-contained in the string that is the second argument.
-The result is `true` if the string matches the iregexp and `false`
-otherwise.
-
-The result is `Nothing` if the first argument is not a string or
-the second argument is not a string conforming to {{-iregexp}}.
+Its arguments are instances of ValueType.
```

Comment:
We should mention that a Singular Path is also acceptable as the first argument.

Reply:
Again, I think "instances of `ValueType` includes these.